src/components/Home.js

```diff
@@ -12,12 +15,50 @@ const Home = (props) => {
     history.push('/ListView');
   };
 
+  const onTokenInputChange = (e) => {
+    setToken(e.target.value);
+  };
+
+  const onSubmitHandler = (e) => {
+    e.preventDefault();
+
+    db.collection(token)
+      .get()
+      .then(function (querySnapshot) {
+        if (querySnapshot.empty) {
+          alert('uh oh! list does not exist');
+        } else {
+          localStorage.setItem('token', token);
+          history.push('/ListView'); // when joining url is /ListView but ListView component does not render until refresh
```

Comment:
We can use the setAuth method which is already part of this component's props to update the auth state of App component. This will alert App that a token has been entered and the user will be able to view the list without refresh.

Reply:
Awesome, thanks for the thorough explanation! Updated the code to set the auth state! https://github.com/the-collab-lab/tcl-16-smart-shopping-list/pull/23/commits/141e77ad7379cd42b6ab683beea8f29f3f505893